lms/services/blackboard_api.py

```diff
@@ -1,6 +1,32 @@
+from marshmallow import INCLUDE
+
+from lms.services.exceptions import HTTPError, OAuth2TokenError
+from lms.validation import RequestsResponseSchema, ValidationError
 from lms.validation.authentication import OAuthTokenResponseSchema
 
 
+class BlackboardErrorResponseSchema(RequestsResponseSchema):
+    """
+    Schema for validating error responses from the Blackboard API.
+
+    This schema's parse() method will return the given response's JSON body as
+    a dict (with whatever fields the response body from Blackboard had) and
+    will default to returning an empty dict if the given response's body isn't
+    a JSON dict, or isn't a dict at all, or if the response isn't a
+    requests.Response object (e.g. if it's None).
+
+    """
+
+    class Meta:
+        unknown = INCLUDE
+
```

Comment:
There's nothing we could assume will be always there? `message`?

Reply:
Hmm, yeah I think I could add `status` and `message` with `required=False` just for documentation purposes. And maybe `fields.Raw` rather than `Int` and `Str`. In trying out this code different ways I found it was simpler if the schema didn't let the calling code make any assumptions about the contents of the dict. If I tried to make the schema enforce that the dict contained a `message` string it just ended up awkward when you try to deal with all the situations (when the JSON body doesn't contain a `message` but does contain other fields, when it does have a `message` but it's not a string, when the JSON body isn't even a dict, ...). But I could add `message = fields.Raw(required=False)` just for documentation purposes

Reply:
Didn't know about fields.Raw, I think that would be useful, the empty schema feels a bit odd to me otherwise (Even If I know what's doing)

Both `required=False` and `allow_none=True` might be needed

Reply:
Added these as raw fields